linux/devdoc/async_socket_linux_requirements.md

```diff
@@ -231,9 +191,7 @@ MOCKABLE_FUNCTION(, ASYNC_SOCKET_SEND_SYNC_RESULT, async_socket_send_async, ASYN
 
   - **SRS_ASYNC_SOCKET_LINUX_11_056: [** `async_socket_send_async` shall create a context for the send where the `payload`, `on_send_complete` and `on_send_complete_context` shall be stored. **]**
```

Comment:
>shall create a context for the send where the `payload`, `on_send_complete` and `on_send_complete_context` shall be stored. [](http://example.com/codeflow?start=67&length=123)

So my understanding is that this "context" will be somehow queued to the completion port's notifications.

What happens when the completion port says "I am ready for more sends" and AT THE SAME TIME `async_socket_send_async` is called from another thread? 

Which wins?

Reply:
I'm not really sure I understand the question.  In async_socket_send_async it creates a new memory allocation with all the data used to send, so the previous call to send shouldn't affect this call.